### ISSUE TITLE
Enable feature nginx + php-fpm website. LE SSL cert without website.

### ISSUE THREAD

**Issue by chris001** [webmin/webmin#550]
* After using `apt` to remove `apache2`, virtualmin still offers mentions Apache under `Enabled features` in the tooltip for `Setup Website for domain` and `Setup SSL website too`.
* Would it be possible to add the ability to have `nginx` + `php-fpm` websites, since this is probably the best combination for low memory servers ie  512MB RAM, due to extremely low memory usage and high speed.
* Would it be possible to have an option for enabling SSL certs for domain, without having to turn on `Setup Website for domain` and `Setup SSL website too`.  Some servers don't run a web server, such as 512MB ram mail servers, where all you need is a LE SSL cert for postfix , dovecot, webmin, usermin, and possibly mysql, or any service.  The idea is, to unlink the LE SSL cert feature from SSL website, and just let it be a cert or more than one cert, that gets used by the various services on the server, whatever they are, no assumptions that you have to be running a web server.

**Comment by jcameron** [comment]
Nginx and FPM are totally supported, however you first need to disable Apache on the Features and Plugins page in Virtualmin, then install the `virtualmin-nginx` and `virtualmin-nginx-ssl` plugins.

However, what we don't yet support is SSL without a website. There isn't even any way to request a Let's Encrypt cert (currently) without a site.

**Comment by chris001** [comment]
Actually, `./certbot-auto certonly --standalone -d mydomain.com,www.mydomain.com,mail.mydomain.com,admin.mydomain.com,ftp.mydomain.com` lets you request an LE cert without a website.  `certbot-auto` script will start its own very basic python web server, for the purpose of responding to the remote calls from the LE servers.  Obviously if apache or nginx is running, then you have to stop them, so that the certbot can listen on the web server ports.

**Comment by jcameron** [comment]
Right, Virtualmin could in theory do the same thing - we just haven't implemented it yet. Also, the SSL to website association in Virtualmin is pretty tight right now.

**Comment by chris001** [comment]
* One thing that's unnecessarily confusing and should be fixed.
Under "System Settings/ Features and Plugins", you have 2 entries for Apache, and 2 entries from Nginx (after you install the 2 additional nginx packages).
The Nginx entries are named right: "Nginx Website", "Nginx SSL Website".
For the Apache entries, "Apache Website", "SSL Website", that's ambiguous. It should be "Apache SSL Website".

* Another bug. When I hit Save on this page - after having disabled the Apache Website and SSL Website, and enabled the Nginx Website and Nginx SSL Website check boxes - I get this error.
```
Failed to save enabled features : Virtualmin is configured to setup DNS zones, but this system is not setup to use itself as a DNS server. Either add 127.0.0.1 to the list of DNS servers, or turn off the BIND feature on the module config page.
```
When I then hit that link to update these settings to how Virtualmin wants them, then, at every reboot, Virtualmin forgets the changes I made, and again shows the same warning about 127.0.0.1 missing from the DNS client settings.  I have to set DNS server (I add 127.0.0.1) and hostname (I change it from just hostname to fully qualified domain name) at every reboot.  The location of this page is "Network Settings / Hostname and DNS Client".

**Comment by jcameron** [comment]
I'll fix that wording.

The second error is legitimate, but admittedly a bit confusing in this case - when you save enabled features, ALL system state is checked which can include errors unrelated to those features.

**Comment by chris001** [comment]
Yes, Virtualmin isn't saving that "Network Settings/ Hostname and DNS Client" settings - something is making it so that the user updated settings for hostname and DNS server are lost when you reboot.

**Comment by jcameron** [comment]
Are you perhaps using DHCP to assign the primary IP address on your system? If so, I'd recommend switching to a static IP.

**Comment by chris001** [comment]
You're right, DHCP is being used, the hosting provider says to use DHCP because their central vm-administration server hands out the IP addresses at boot up, so that in case the vm's IP address ever needs to change to something else (which it hasn't in over a year), their DHCP system will handle it with no loss of uptime and no need for each server customer to manually edit each vm.  The vm's are fully kernel virtualized mode, so they're not accessible from the outside by the hosting provider.  If a static IP were to be set, then the vm might become impossible to connect to, except possibly through the special ssh tty terminal mode.

**Comment by chris001** [comment]
This issue is also happening (it forgets the DNS client setting 127.0.0.1) on a virtualmin server using static IP.

**Comment by jcameron** [comment]
I'd strongly recommend not using DHCP - if your IP does change, it would break all your DNS records.

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.
